### PR TITLE
Make paint function accept anything implementing Surface trait

### PR DESCRIPTION
**Pull request by JerzySpendel**
Closes <https://github.com/emilk/egui/issues/680>.

### CODE REVIEW

File: egui_glium/src/lib.rs [hunk 1]
```diff
@@ -580,10 +580,10 @@ impl EguiGlium {
         (needs_repaint, shapes)
     }
 
-    pub fn paint(
+    pub fn paint<T: glium::Surface>(
         &mut self,
         display: &glium::Display,
-        target: &mut glium::Frame,
+        target: &mut T,
```
Comment (emilk) [inline egui_glium/src/lib.rs:586]: Let's replace generics with dynamic dispatch instead to keep down own monomorphization overhead (larger code and slower compilation).

```suggestion
        target: &mut dyn glium::Surface,
```

here and below

**Review by emilk — CHANGES_REQUESTED** [review]

File: egui_glium/src/lib.rs (hunk 1 above)
Comment (JerzySpendel) [reply egui_glium/src/lib.rs:586]: He @emilk, thanks for reply :) That was also my thought to make it dynamic dispatch, but it's not possible as Surface trait is not object-safe.

Comment (emilk) [reply egui_glium/src/lib.rs:586]: Oh, ok then :)

**Comment by emilk** [comment]
You got some unused import warnings, then LGTM

**Comment by JerzySpendel** [comment]
Done

**Pull request merged**